server/scheduler.go

```diff
@@ -297,24 +298,42 @@ func scheduleRemovePeer(cluster *clusterInfo, s Selector, filters ...Filter) (*r
 
 // scheduleTransferLeader schedules a region to transfer leader to the peer.
 func scheduleTransferLeader(cluster *clusterInfo, s Selector, filters ...Filter) (*regionInfo, *metapb.Peer) {
-	sourceStores := cluster.getStores()
-
-	source := s.SelectSource(sourceStores, filters...)
-	if source == nil {
+	stores := cluster.getStores()
+	if len(stores) == 0 {
 		return nil, nil
 	}
 
-	region := cluster.randLeaderRegion(source.GetId())
-	if region == nil {
-		return nil, nil
+	var averageLeader float64
+	for _, s := range stores {
+		averageLeader += float64(cluster.getStoreLeaderCount(s.GetId())) / float64(len(stores))
```

Comment:
In fact I intentionally avoid using s.leaderCount() here, because leader count in cache is more up to date.